### PR TITLE
Adds Sidecar API

### PR DESCRIPTION
**Pull request by albertov19**
### Description

This PR adds the Sidecar API page. Still missing an image and to check Substrate Sidecar API Types
### Checklist

- [x] If this page requires a disclaimer, I have added one
- [ ] If pages have been moved around, I have created an additional PR in `moonbeam-mkdocs` to update redirects

### CODE REVIEW

File: builders/interact/substrate-api/sidecar-api.md [hunk 1]
```diff
@@ -0,0 +1,495 @@
...
+
+```typescript
+--8<-- 'code/substrate-api/sidecar-transfer.ts'
+```
+
+### ERC-20 Token Transfers {: #erc-20-token-transfers }
+
+Events emitted by smart contracts such as an ERC-20 token contract deployed on EVM ContainerChains can be decoded from Sidecar block JSON objects. The nesting structure is as following:
+
+```JSON
```
Comment (augustocollado) [inline builders/interact/substrate-api/sidecar-api.md:288]:
```suggestion
```
```

Comment (augustocollado) [reply builders/interact/substrate-api/sidecar-api.md:288]: I'd delete the JSON type of code. It renders the text in a funny way.
(Applies to every tree type structure description in the doc)

File: builders/interact/substrate-api/sidecar-api.md [hunk 2]
```diff
@@ -0,0 +1,495 @@
...
+
+Although EVm ContainerChains are Ethereum-compatible, they are also Substrate-based chains at its core, and priorities work differently in Substrate than in Ethereum. In Substrate, transactions are not prioritized by gas price. To address this, EVM ContainerChains uses a modified prioritization system that reprioritizes Substrate transactions using an Ethereum-first solution. A Substrate transaction still goes through the validity process, where it is assigned transaction tags, longevity, and a priority. The original priority is then overwritten with a new priority based on the transaction's fee per gas, which is derived from the transaction's tip and weight. If the transaction is an Ethereum transaction, the priority is set according to the priority fee.
```
Comment (augustocollado) [inline builders/interact/substrate-api/sidecar-api.md:451]:
```suggestion
Although EVM ContainerChains are Ethereum-compatible, they are also Substrate-based chains at their core, and priorities work differently in Substrate and in Ethereum. In Substrate, transactions are not prioritized by gas price. To address this, EVM ContainerChains uses a modified prioritization system that reprioritizes Substrate transactions using an Ethereum-first solution. A Substrate transaction still goes through the validity process, where it is assigned transaction tags, longevity, and priority. The original priority is then overwritten with a new priority based on the transaction's fee per gas, which is derived from the transaction's tip and weight. If the transaction is an Ethereum transaction, the priority is set according to the priority fee.
```

File: builders/interact/substrate-api/polkadot-js-api.md [hunk 3]
```diff
@@ -257,6 +257,31 @@ console.log(`Submitted with hash ${txHash}`);
 
 Note that the `signAndSend` function can also accept optional parameters, such as the `nonce`. For example, `signAndSend(alice, { nonce: aliceNonce })`. You can use the [sample code from the State Queries](#state-queries){target=_blank} section to retrieve the correct nonce, including transactions in the mempool.
 
+### Fee Information {: #fees}
+
+The transaction endpoint also offers a method to obtain weight information for a given `api.tx.<module>.<method>`. To do so, you'll need to use the `paymentInfo` function after having built the entire transaction with the specific `module` and `method`.
+
+The `paymetnInfo` function returns weight information in terms of `refTime` and `proofSize`, which can be used to determined the transaction fee. This is extermely helpful when crafting remote execution calls via XCM.
```
Comment (augustocollado) [inline builders/interact/substrate-api/polkadot-js-api.md:264]:
```suggestion
The `paymetnInfo` function returns weight information in terms of `refTime` and `proofSize`, which can be used to determine the transaction fee. This is extremely helpful when crafting remote execution calls via XCM.
```

**Review by augustocollado — CHANGES_REQUESTED** [review]
Great!
I left some comments and minor typo corrections

File: builders/interact/substrate-api/polkadot-js-api.md (hunk 3 above)
Comment (eshaben) [reply builders/interact/substrate-api/polkadot-js-api.md:264]: doing a drive by 🚗 `paymentInfo` is a typo too!

File: builders/interact/substrate-api/sidecar-api.md [hunk 4]
```diff
@@ -0,0 +1,495 @@
...
+## Setting up the Substrate API Sidecar {: #setting-up-the-substrate-api-sidecar }
+
+In the terminal that Sidecar will run, export the environmental variable for the WS endpoint of the network. For example: 
+
+=== "Dancebox"
+    ```
+    export SAS_SUBSTRATE_URL=wss://fraa-dancebox-rpc.a.dancebox.tanssi.network
+    ```
+
+=== "Dancebox EVM ContainerChain"
```
[the remark this answers is not in the document]
Comment (albertov19) [reply builders/interact/substrate-api/sidecar-api.md:44]: I don't think this is necessary. I'm not sure how the endpoint structure is going to be. I can add a note saying that you should replace this with your ContainerChain endpoint

File: builders/interact/substrate-api/sidecar-api.md [hunk 5]
```diff
@@ -0,0 +1,495 @@
...
+To install the Substrate API Sidecar service locally in the current directory, run this from the command line:
+
+```
+npm install @substrate/api-sidecar@{{ networks.dancebox.substrate_api_sidecar.stable_version }}
+```
+
+!!! note
+    If the current folder does not already have a Node.js project structure, you need to manually created the `node_modules` directory by typing `mkdir node_modules`.
+
+Substrate API Sidecar v{{ networks.dancebox.substrate_api_sidecar.stable_version }} is the current stable version that has been tested to work with ContainerChains. You can verify the installation was successful by typing from the installation directory root:
```
[the remark this answers is not in the document]
Comment (albertov19) [reply builders/interact/substrate-api/sidecar-api.md:29]: Our tests with Moonbeam are using V14.1.1 - But we can try bumping this to 17.x and check

**Review by augustocollado — APPROVED** [review]
lgtm!

**Pull request merged**